Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayTestHelper.cs

```diff
@@ -0,0 +1,300 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Amazon.APIGateway;
+using Amazon.APIGateway.Model;
+using Amazon.ApiGatewayV2;
+using Amazon.ApiGatewayV2.Model;
+using Amazon.IdentityManagement;
+using Amazon.IdentityManagement.Model;
+using Amazon.Lambda;
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Model;
+using Amazon.Lambda.TestTool.Extensions;
+using Amazon.Lambda.TestTool.Models;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+
+namespace Amazon.Lambda.TestTool.IntegrationTests
+{
+    public class ApiGatewayTestHelper
+    {
+        private readonly IAmazonAPIGateway _apiGatewayV1Client;
+        private readonly IAmazonApiGatewayV2 _apiGatewayV2Client;
+        private readonly IAmazonLambda _lambdaClient;
+        private readonly IAmazonIdentityManagementService _iamClient;
+        private readonly HttpClient _httpClient;
+
+        public ApiGatewayTestHelper(
+            IAmazonAPIGateway apiGatewayV1Client,
+            IAmazonApiGatewayV2 apiGatewayV2Client,
+            IAmazonLambda lambdaClient,
+            IAmazonIdentityManagementService iamClient)
+        {
+            _apiGatewayV1Client = apiGatewayV1Client;
+            _apiGatewayV2Client = apiGatewayV2Client;
+            _lambdaClient = lambdaClient;
+            _iamClient = iamClient;
+            _httpClient = new HttpClient();
+        }
+
+        public async Task<string> CreateLambdaFunctionAsync(string roleArn, string lambdaCode)
```

Comment:
Please create the function with a CloudFormation template so it is easier to tear down when we have orphan test data when tests crash.

Reply:
i would like to update it to use cloudformation in a separate PR, after i get the main functionality in.

Reply:
actually, i figured out how to do this. i will push an update using cloudformation

Reply:
ive updated it to use cloudformation to create everything. i reran the integration tests too and they all pass (except for the 1 i have intentionally failing right now related to the v2 json assumptions)